project/frontend/src/components/PreferenceForm.js

```diff
@@ -24,17 +28,34 @@ class PreferenceForm extends React.Component {
         lng: '',
       },
       open: true,
+      cuisineOptions: [],
     };
     this.changeState = this.changeState.bind(this);
     this.handleSubmit = this.handleSubmit.bind(this);
   }
 
+  componentWillUpdate() {
+    const { currLocation } = this.state;
+    if (
+      this.props.currLocation.lat &&
+      this.props.currLocation.lng &&
+      (currLocation['lat'] !== this.props.currLocation.lat ||
+        currLocation['lng'] !== this.props.currLocation.lng)
+    ) {
+      currLocation['lat'] = this.props.currLocation.lat;
+      currLocation['lng'] = this.props.currLocation.lng;
```

Comment:
hm this is modifying state directly? might want to pull these out to const newLat = this.props.currLocation.lat

Reply:
(not actually modifying state directly, but it's slightly confusing - totally optional though)

Reply:
So are you suggesting then setting `currLocation['lat'] = newLat`?